File: GenerateMd5/GenerateMd5.dart
import 'dart:convert';
import 'dart:typed_data';
import "package:pointycastle/export.dart";
import 'package:convert/convert.dart';

main() {
/* add in pubspec.yaml:
dependencies:
  pointycastle: ^3.1.1
 */

  printC('Generate a MD5 hash');

  printC('\n# # # SECURITY WARNING: This code is provided for achieve    # # #');
  printC('# # # compatibility between different programming languages. # # #');
  printC('# # # It is NOT SECURE - DO NOT USE THIS CODE ANY LONGER !   # # #');
  printC('# # # The hash algorithm MD5 is BROKEN.                      # # #');
  printC('# # # DO NOT USE THIS CODE IN PRODUCTION                     # # #\n');

  final plaintext = 'The quick brown fox jumps over the lazy dog';
  printC('plaintext: ' + plaintext);

  Uint8List md5Value = calculateMd5(plaintext);
  printC('\nmd5Value (hex) length: ' + md5Value.length.toString() + ' data: ' + bytesToHex(md5Value));
}

Uint8List calculateMd5(String data) {
  var dataToDigest = createUint8ListFromString(data);
  var d = Digest('MD5');
  return d.process(dataToDigest);
}

String bytesToHex(Uint8List data) {
  return hex.encode(data);
}

Uint8List createUint8ListFromString(String s) {
  var ret = new Uint8List(s.length);
  for (var i = 0; i < s.length; i++) {
    ret[i] = s.codeUnitAt(i);
  }
  return ret;
}

void printC(String newString) {
  // for compatibility reasons to FlutterEmptyConsole
  print(newString);
}

File: GenerateMd5/GenerateMd5.go
package main

import (
  "crypto/md5"
  "encoding/hex"
	"fmt"
  "strconv"
)

func main() {

  fmt.Println("Generate a MD5 hash")

  fmt.Println("\n# # # SECURITY WARNING: This code is provided for achieve    # # #")
  fmt.Println("# # # compatibility between different programming languages. # # #")
  fmt.Println("# # # It is NOT SECURE - DO NOT USE THIS CODE ANY LONGER !   # # #")
  fmt.Println("# # # The hash algorithm MD5 is BROKEN.                      # # #")
  fmt.Println("# # # DO NOT USE THIS CODE IN PRODUCTION                     # # #\n")  

  plaintext := string("The quick brown fox jumps over the lazy dog")
  fmt.Println("plaintext:                " + plaintext)

  md5Value := CalculateMd5(plaintext)
  md5Len := len(md5Value)
  md5String := hex.EncodeToString(md5Value)  
  fmt.Println("md5Value (hex) length: " + strconv.Itoa(md5Len) + " data: " + md5String)
}

func CalculateMd5(string string)([]byte){
  dataToHash := []byte(string)
  msgHash := md5.New()
  _, err := msgHash.Write(dataToHash)
  if err != nil {
	  panic(err)
  }
  msgHashSum := msgHash.Sum(nil)
  return msgHashSum
}
